File: Тренировка 1.0/02. Линейный поиск/E. Чемпионат по метанию коровьих лепешек/Чемпионат по метанию коровьих лепешек.cs
using System;

int n = int.Parse(Console.ReadLine());
string[] results = Console.ReadLine().Split(' ');

int bestThrowIndex = 0;
short bestThrow = short.Parse(results[0]);
for (int i = 1; i < n; i++)
{
    short number = short.Parse(results[i]);
    if (bestThrow < number)
    {
        bestThrow = number;
        bestThrowIndex = i;
    }
}

short vasyaThrow = -1;
for(int i = bestThrowIndex + 1; i < n - 1; i++)
{
    short throwResult = short.Parse(results[i]);
    if (throwResult % 10 == 5 && throwResult > short.Parse(results[i + 1]) && vasyaThrow < throwResult) vasyaThrow = throwResult;
}

if (vasyaThrow == -1) Console.WriteLine(0);
else
{
    int position = 1;
    for (int i = 0; i < n; i++)
        if (vasyaThrow < short.Parse(results[i])) position++;

    Console.WriteLine(position);
}

File: Тренировка 1.0/02. Линейный поиск/E. Чемпионат по метанию коровьих лепешек/Чемпионат по метанию коровьих лепешек.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int n;
    cin >> n;

    vector<short> results;
    int bestThrowIndex = 0;
    short bestThrow = -1;
    for (int i = 0; i < n; i++)
    {
        short number;
        cin >> number;
        results.push_back(number);

        if (number > bestThrow)
        {
            bestThrow = number;
            bestThrowIndex = i;
        }
    }

    short vasyaThrow = -1;
    for (int i = bestThrowIndex + 1; i < n - 1; i++)
    {
        short throwResult = results[i];
        if (throwResult % 10 == 5 && throwResult > results[i + 1] && vasyaThrow < throwResult) vasyaThrow = throwResult;
    }

    if (vasyaThrow == -1) cout << 0;
    else
    {
        int position = 1;
        for (int i = 0; i < n; i++)
            if (vasyaThrow < results[i]) position++;

        cout << position;
    }
}